Web Quiz Engine (Kotlin)/task/src/engine/ControllerAdvice.kt

```diff
@@ -1,33 +1,21 @@
 package engine
 
+import engine.exception.QuizAnswerIndexException
+import engine.exception.QuizNotFoundException
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.MissingServletRequestParameterException
 import org.springframework.web.bind.annotation.ControllerAdvice
 import org.springframework.web.bind.annotation.ExceptionHandler
 
-class QuizAnswerIndexException(private val faultyIndex: Int): RuntimeException() {
-    override val message: String
-        get() = "The answer index $faultyIndex is out of bounds!"
-}
-
-class QuizNotFoundException(private val quizId: Int): RuntimeException() {
-    override val message: String
-        get() = "Quiz with id $quizId not found!"
-}
-
 @ControllerAdvice
 class ControllerAdvice {
 
-    @ExceptionHandler(MissingServletRequestParameterException::class)
-    fun handleMissingRequestParameterException(e: MissingServletRequestParameterException) =
-        ResponseEntity(mapOf("error" to e.message), HttpStatus.BAD_REQUEST)
-
-    @ExceptionHandler(QuizAnswerIndexException::class)
-    fun handleInvalidQuizAnswerIndexException(e: QuizAnswerIndexException) =
+    @ExceptionHandler(MissingServletRequestParameterException::class, QuizAnswerIndexException::class)
+    fun handleException(e: Exception) =
```

Comment:
I think that's how it should be but I'm not a 100% sure.

Reply:
It is typically better to define an Error payload object to populate and return it in these handlers, but for simple cases like these returning a map should suffice.

Reply:
Alright, thks for the new knowledge. No change. Done